common/src/main/java/com/linecorp/centraldogma/internal/Util.java

```diff
@@ -179,6 +179,19 @@ public static String validateEmailAddress(String emailAddr, String paramName) {
                 " (expected: " + EMAIL_PATTERN.pattern() + " or IP address domain)");
     }
 
+    public static String ensureEmailAddress(String emailAddr, String paramName) {
+        requireNonNull(emailAddr, paramName);
+        if (isValidEmailAddress(emailAddr)) {
+            return emailAddr;
+        }
+        return emailAddr + "@localhost.localdomain";
+    }
+
+    public static String parseLogin(String emailAddr, String paramName) {
```

Comment:
How about `emailToUsername`? It doesn't seem to parse but just extract in my opinion.